connection.js

```diff
@@ -0,0 +1,65 @@
+'use strict'
+
+const fp = require('fastify-plugin')
+
+function fastifyMysql (fastify, options, next) {
+  const name = options.name
+  delete options.name
+
+  const usePromise = options.promise
+  delete options.promise
+
+  const db = {}
+  if (usePromise) {
+    const mysql = require('mysql2/promise')
+    mysql.createConnection(options.connectionString || options)
+      .then((connection) => {
+        fastify.addHook('onClose', (fastify, done) => connection.end(done))
+        db.connection = connection
+        db.query = connection.query.bind(connection)
+        // synchronous functions
+        db.format = connection.format.bind(connection)
+        db.escape = connection.escape.bind(connection)
+        db.escapeId = connection.escapeId.bind(connection)
+        next()
+      })
+      .catch((e) => {
+        next(e)
+      })
```

Comment:
you should return the promise, not call `next()` in this case.

Reply:
I need its connection for binding first. If I return the promise not wait for calling next(), it will hang on there.